pkg/timestream/datasource_test.go

```diff
@@ -25,11 +25,22 @@ type fakeRunner struct {
 
 func (f *fakeRunner) runQuery(ctx context.Context, input *timestreamquery.QueryInput) (*timestreamquery.QueryOutput, error) {
 	res := &timestreamquery.QueryOutput{}
+	dimensions := []*timestreamquery.Datum{}
```

Comment:
This is adding a lot of behavior a bit hidden away here in the fakeRunner. From what I understand the result needs to be populated with very specific lengths of rows, row Data, ArrayValue, etc.
Would it be possible to simply mock that return result in the specific dimensions test rather than build it in here? If that's the same, it would be a lot clearer (more explicit way to generate that result) and impact no other test.